kbchat/kvstore_test.go

```diff
@@ -0,0 +1,60 @@
+package kbchat
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// test doesn't make assumption about prior state
+func TestKVStore(t *testing.T) {
+	alice, dir := testBotSetup(t, "alice")
+	channel := getTeamChatChannel(t, "acme")
+	team := channel.Name
+	defer testBotTeardown(t, alice, dir)
+
+	namespace := "_test_namespace1"
+	key := "_test_key1"
+
+	// put with default revision
+	res2, err2 := alice.PutEntry(team, namespace, key, "value1", 0)
+	require.NoError(t, err2)
+	rev := res2.Revision
+
+	// fail put
+	_, err3 := alice.PutEntry(team, namespace, key, "value2", rev)
+	require.Error(t, err3)
+
+	// list namespaces
+	res4, err4 := alice.ListNamespaces(team)
+	require.NoError(t, err4)
+	require.True(t, len(res4.Namespaces) > 0)
+
+	// list entryKeys
+	res5, err5 := alice.ListEntryKeys(team, namespace)
+	require.NoError(t, err5)
+	require.True(t, len(res5.EntryKeys) > 0)
+
+	// get
+	res6, err6 := alice.GetEntry(team, namespace, key)
+	require.NoError(t, err6)
+	require.Equal(t, "value1", res6.EntryValue)
+
+	// fail delete
+	_, err7 := alice.DeleteEntry(team, namespace, key, rev+2)
+	require.Error(t, err7)
+
+	// delete
+	res8, err8 := alice.DeleteEntry(team, namespace, key, rev+1)
```

Comment:
Huh, I'm missing something -- if `rev` came back from the successful put, and that's the last operation that succeeded, why do we pass `rev+1` to get a successful delete, rather than just `rev`?

Reply:
Ah!  I figured it out, I wasn't expecting `delete` to take an *anticipated* revision rather than the last revision you're aware of, but `delete` and `put` both do that.  Do we have a doc that describes those semantics of `revision` yet?

Reply:
yeah this is an interesting point. it's sort of implied by the output of `keybase kvstore api --help`:
```
(specifying a non-zero revision enables custom concurrency behavior, e.g. 1 will throw an error if the entry already exists)
```
but you're right that this isn't nearly as clear and obvious as it could be. fwiw it's our assumption that few usecases will actually require specifying the revision. if you leave it blank, the client will fetch the entry, and increment the revision for you. it's only if you might need to know what's been written previously that you'll need to specify a revision.